docs/calendar-draft.md

```diff
@@ -0,0 +1,315 @@
+# Draft Design of Temporal Calendar API
+
+This doc describes a design for first-class support for non-Gregorian [calendars](https://en.wikipedia.org/wiki/Calendar) in Temporal.
+
+## Temporal.Date internal slots
+
+Temporal.Date currently has three internal slots: year, month, and day. (An "internal slot" refers to actual data, as opposed to "properties", which could be computed.)  In this proposal, three more internal slots would be added:
+
+1. calendar
+2. isLeapMonth
+3. era
+
+It is believed that a date in any known calendar can be described using the five-tuple (*year*, *month*, *day*, *isLeapMonth*, *era*).  It is possible that *isLeapMonth* can be merged into *month* by expanding the set of numbers that *month* can represent.
```

Comment:
It would be good state in this document that we can't represent calendars that don't fit into this tuple.

Reply:
I think the idea was actually that a Calendar could store whatever values it wants into the slots.  Added a clarification.